File: src/2283.check-if-number-has-equal-digit-count-and-digit-value.2377/solution.ts
function digitCount (num: string): boolean {
    const counts = new Array<number>(10).fill(0);
    for (const n of num) {
        counts[n]++;
    }
    for (let i = 0; i < num.length; i++) {
        if (num[i] !== String(counts[i])) {
            return false;
        }
    }
    return true;
}

File: src/2283.check-if-number-has-equal-digit-count-and-digit-value.2377/solution.java
class Solution {
    public boolean digitCount(String num) {
        int[] counts = new int[10];
        char[] chars = num.toCharArray();
        for (char c : chars) {
            counts[c - '0']++;
        }
        for (int i = 0; i < chars.length; i++) {
            if (counts[i] != chars[i] - '0') {
                return false;
            }
        }
        return true;
    }
}